### ISSUE TITLE
feat : Remote OCA bundle resolver

### ISSUE THREAD

**Issue by cvarjao** [openwallet-foundation/bifold-wallet#676]
Support for OCA bundle files stored remotely (via http)

Acceptance Criteria:
- Given an base URL (e.g.: `https://raw.githubusercontent.com/bcgov/aries-oca-bundles/main/OCABundles/`), the wallet should be able to discover/download/cache OCA bundle files as needed. Each wallet implementation may provide a different base URL.
- A wallet implementation (e.g.: BC Wallet) may provide a list of bundle IDs(?) to warm up cache during app initialization

```
GET /all.json
  {
    "<schema_id> or <cred_def_id>":{},
    "<schema_id> or <cred_def_id>":{},
  }

GET /<schema_id>/bundle.json
  {
     "captureBase": {},
     "overlays": [
        {"capture_base": ""}
      ]
  }

GET /<schema_id>/<cred_def_id>/bundle.json
  {
     "captureBase": {},
     "overlays": [
        {"capture_base": ""}
      ]
  }
```

**Comment by thiagoromanos** [comment]
I would sugest the [GET] /all.json to return a map instead of a array, so the resolver knows what it's getting.

**Comment by thiagoromanos** [comment]
Just opened a draft for the remote OCA bundle resolver and the extraction of the current bifold OCA bundle resolver.

**Comment by cvarjao** [comment]
@thiagoromanos , some related conversation in bcgov/aries-oca-bundles#22 . @swcurran is suggesting an index file that is cacheable for quick lookup. Any thoughts on that?

**Comment by swcurran** [comment]
Here is what the generated file is like:

https://github.com/bcgov/aries-oca-bundles/blob/main/ocabundles.json

**Comment by thiagoromanos** [comment]
I think the index file is a great idea. That doesn't excludes the "/all.json" endpoint, right? (fetch each bundle at startup would be slower than fetching a single larger file)

**Comment by thiagoromanos** [comment]
> Here is what the generated file is like:
> 
> https://github.com/bcgov/aries-oca-bundles/blob/main/ocabundles.json

The remote resolver is implemented and merged into bifold (#891). There is still room for upgrades, but it is able to read the index file mentioned by @swcurran and load bundles from it (just like oca explorer).
